lib/rubocop/cop/factory_bot/association_style.rb

```diff
@@ -218,6 +224,14 @@ def options_for_autocorrect_to_implicit_style(node)
           end
           options
         end
+
+        def trait_within_trait?(node)
+          factory_node = node.each_ancestor(:block).find do |ancestor|
```

Comment:
What if it’s a factory inside a factory and the trait used in the inner one is defined in the outer one?

Reply:
Indeed it is, I updated this PR to take into account the case of factory inside a factory.